### ISSUE TITLE
Add no-crop thumbnails rendering in images preview

### ISSUE THREAD

**Issue by pasnox** [owncloud/ocis#5179]
## Describe the bug

Image preview is broken.
It regularly produce thumbnails that truncate most of the images and make the previewing in Space useless.

## Steps to reproduce

1. Create a space
2. Upload a bunch of landscape and portrait photos
3. Preview them, in most occasion the thumbnail produced is truncated

## Expected behavior
The image is always entirely visible, downscaling it if needed to fit in the client screen

## Actual behavior
Thumbnails seems to be cropped instead of downscaled

## Setup

<details>
<p>

```console
OCIS_URL="https://host.com:8888" \
OCIS_CONFIG_DIR="${HOME}/ocis/config" \
OCIS_BASE_DATA_PATH="${HOME}/ocis/data" \
OCIS_INSECURE=false \
PROXY_TRANSPORT_TLS_KEY="/etc/pki/host.com/private/csr.key" \
PROXY_TRANSPORT_TLS_CERT="/etc/pki/host.com/certificates/certificate-fullchain.crt" \
PROXY_HTTP_ADDR="0.0.0.0:8888" \
WEB_HTTP_ADDR="0.0.0.0:8431" \
ocis server
```

</p>
</details>

**Comment by pasnox** [comment]
So, after some investigation, it seems the preview thumbnails used depend the client resolution and how works the thumbnailer by default.
So the thumbnailer by default only generate landscape *cropped* thumbnails, so if the client size is lower that the image size, you would get a highly truncated image.
To workaround (but not fix it), you can request the thumbnailer to thumbnails also in portrait mode, extending the default

from:
*THUMBNAILS_RESOLUTIONS="16x16,32x32,64x64,128x128,1920x1080,3840x2160,7680x4320"*

to:
*THUMBNAILS_RESOLUTIONS="16x16,32x32,64x64,128x128,1920x1080,3840x2160,7680x4320,1080x1920,2160x3840,4320x7680"* 

Please add those portrait mode in default, and ensure downscaled instead of cropped is used by the thumbnailer.

**Comment by micbar** [comment]
@C0rby Can you take a look if this is a bug?

**Comment by kulmann** [comment]
There is a ticket in web for this, see https://github.com/owncloud/web/issues/7728 - but we need some kind of `nocrop` query option for the thumbnailer service, so that web can request a preview cropped or downscaled as needed.

We intended to have a cropped thumbnail for the file list, but downscaled for the right sidebar `Details` panel, for the `preview` app (which shows the images in larger dimensions) and for the upcoming tiles view.

**Comment by C0rby** [comment]
> Please add those portrait mode in default

You are more than welcome to add the missing resolutions to [the default config](https://github.com/owncloud/ocis/blob/4b18930944111fe9ba5916d1fedc4f1bf38a143a/services/thumbnails/pkg/config/defaults/defaultconfig.go#L40) and open a PR.

> and ensure downscaled instead of cropped is used by the thumbnailer.

As @kulmann already stated that was a deliberate decision but we could add a `nocrop` query parameter.
For that one would need to:
- adapt the webdav service https://github.com/owncloud/ocis/blob/4b18930944111fe9ba5916d1fedc4f1bf38a143a/services/webdav/pkg/dav/requests/thumbnail.go#L53
- Add a `nocrop` parameter to the thumbnails service https://github.com/owncloud/ocis/blob/b8a90b5335b3d825298c9a6eaaddac84260e4252/protogen/proto/ocis/services/thumbnails/v0/thumbnails.proto#L41
- Add the parameter to all the requests in https://github.com/owncloud/ocis/blob/master/services/webdav/pkg/service/v0/service.go
- And implement the parameter in the thumbnails service, probably somewhere in the generator file: https://github.com/owncloud/ocis/blob/master/services/thumbnails/pkg/thumbnail/generator.go

https://pkg.go.dev/github.com/disintegration/imaging#Resize could be used for that since it doesn't crop as opposed to https://pkg.go.dev/github.com/disintegration/imaging#Thumbnail, which we are using right now.

**Comment by jacobgkau** [comment]
> Removing the bug label because this is an enhancement.

This is a blocking issue for resolving a bug (impossible to view a full photo without downloading) in web.

**Comment by pasnox** [comment]
Hi, I just migrated to 4.0.4, and it seems not fixed.
I still have cropped landscape images. Also, you dd not said how I would enable the no-crop mode ?
Can you please give more information ?
Thanks,

**Comment by kulmann** [comment]
> Hi, I just migrated to 4.0.4, and it seems not fixed.
> I still have cropped landscape images. Also, you dd not said how I would enable the no-crop mode ?
> Can you please give more information ?
> Also, just found [owncloud/web#9806](https://github.com/owncloud/web/pull/9806) claiming to fix it, but does not seems, or not yet in a released version ?
> Thanks,

It will be available as part of the ocis `v5.0.0` release early next year. There are alpha versions available if you want to check - the web ui utilizes the new option automatically in the `preview` app. First beta is planned for next Tuesday.
